Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/SessionManager.cs

```diff
@@ -0,0 +1,402 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    public struct SessionPlayerData
+    {
+        public ulong ClientID;
+        public string ClientGUID;
+        public string PlayerName;
+        public Vector3 PlayerPosition;
+        public Vector3 PlayerRotation;
+        public NetworkGuid AvatarNetworkGuid;
+        public int CurrentHP;
+        public bool IsConnected;
+        public bool IsReconnecting;
+
+        public SessionPlayerData(ulong clientID, string clientGUID, string name, Vector3 position, Vector3 rotation, NetworkGuid avatarNetworkGuid, int currentHP = 0, bool isConnected = false, bool isReconnecting = false)
+        {
+            ClientID = clientID;
+            ClientGUID = clientGUID;
+            PlayerName = name;
+            PlayerPosition = position;
+            PlayerRotation = rotation;
+            AvatarNetworkGuid = avatarNetworkGuid;
+            CurrentHP = currentHP;
+            IsConnected = isConnected;
+            IsReconnecting = isReconnecting;
+        }
+
+        public SessionPlayerData(ulong clientID, string clientGUID, string name, Vector3 position, Vector3 rotation, int currentHP = 0, bool isConnected = false, bool isReconnecting = false)
+        {
+            ClientID = clientID;
+            ClientGUID = clientGUID;
+            PlayerName = name;
+            PlayerPosition = position;
+            PlayerRotation = rotation;
+            AvatarNetworkGuid = new NetworkGuid();
+            CurrentHP = currentHP;
+            IsConnected = isConnected;
+            IsReconnecting = isReconnecting;
+        }
+    }
+
+    public class SessionManager : MonoBehaviour
+    {
+        public static SessionManager Instance { get; private set; }
+
+        [SerializeField]
+        private GameNetPortal m_Portal;
+
+        /// <summary>
+        /// Maps a given client guid to the data for a given client player.
+        /// </summary>
+        private Dictionary<string, SessionPlayerData> m_ClientData;
+
+        /// <summary>
+        /// Map to allow us to cheaply map from guid to player data.
+        /// </summary>
+        private Dictionary<ulong, string> m_ClientIDToGuid;
+
+        // used in ApprovalCheck. This is intended as a bit of light protection against DOS attacks that rely on sending silly big buffers of garbage.
+        private const int k_MaxConnectPayload = 1024;
```

Comment:
Seems like this was brought over from ServerGameNetPortal and could be removed.

Reply:
Yes you're right, I missed this one, thanks